shared-module/tilepalettemapper/TilePaletteMapper.c

```diff
@@ -0,0 +1,72 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2025 Tim Cocks for Adafruit Industries
+//
+// SPDX-License-Identifier: MIT
+#include <stdlib.h>
+#include "py/runtime.h"
+#include "shared-bindings/tilepalettemapper/TilePaletteMapper.h"
+#include "shared-bindings/displayio/Palette.h"
+
+void common_hal_tilepalettemapper_tilepalettemapper_construct(tilepalettemapper_tilepalettemapper_t *self,
+    mp_obj_t palette, uint16_t width, uint16_t height) {
+
+    self->palette = palette;
+    self->width_in_tiles = width;
+    self->height_in_tiles = height;
+    self->needs_refresh = false;
+    int mappings_len = width * height;
+    self->tile_mappings = (uint16_t **)m_malloc(mappings_len * sizeof(uint16_t *));
+    uint32_t palette_len = common_hal_displayio_palette_get_len(self->palette);
```

Comment:
Ah! Let's take in the number of input indices and then adjust the value length based on the number of colors in the palette.